packages/moonstone/VirtualList/VirtualListBase.js

```diff
@@ -535,15 +529,25 @@ const VirtualListBaseFactory = (type) => {
 							}
 
 						}
-					} else if (
-						directions.right && repeat ||
-						directions.left && Spotlight.move(direction) ||
-						(directions.up || directions.down) && (repeat || moveFocusStraight({id: this.props.spotlightId, direction}) && currentTarget.contains(Spotlight.getCurrent()))
-					) {
-						ev.preventDefault();
-						ev.stopPropagation();
-					} else if (!repeat) {
-						isLeaving = true;
+					} else {
+						const spottables = Spotlight.getSpottableDescendants(spotlightId);
+						const spottablesTotal = spottables.length;
+						const firstScrollButton = spottables[spottablesTotal - 2];
+						const lastScrollButton = spottables[spottablesTotal - 1];
+						const isLastScrollButton = target === lastScrollButton;
+
+						if (
+							directions.right && repeat ||
+							directions.left && Spotlight.move(direction) ||
+							(directions.up && !isLastScrollButton || directions.down && isLastScrollButton) && repeat ||
+							directions.down && !isLastScrollButton && Spotlight.focus(lastScrollButton) ||
+							directions.up && isLastScrollButton && Spotlight.focus(firstScrollButton)
```

Comment:
There is the same logic in `ScrollButtons` in https://github.com/enactjs/enact/blob/develop/packages/moonstone/Scrollable/ScrollButtons.js#L212. If 5-way navigation in scroll buttons is handled in VirtualList, we need to remove the logic in `ScrollButtons` and need to add the logic in Scroller, too.

Reply:
I think the logic in `ScrollButtons` still applies. We only need to apply similar logic in `VirtualList`, since we are preventing the originating keydown events from reaching `Scroller`/`ScrollButtons`. It's perhaps less than ideal, but needed since we explicitly need to manage keydown event bubbling. I think keeping the original logic in `ScrollButtons` makes sense since we do not need to verify and prevent event bubbling there.